File: magic.cpp
#include <stdio.h>

#define TRUE 1
#define FALSE 0

int main()
{
	FILE* fin = stdin;
	int x, y;
	long o, answer, a;
	
	bool possible;
	
	for(;;)
	{
		fscanf(fin, " %d %d", &x, &y);
		
		if(x == 0 && y == 0)
			break;
		
		answer = -1;
		possible = TRUE;
		
		if(x == 0)
			possible = FALSE;
		
		for(o = 1; possible && o <= 10; ++o)
		{
			a = (o * x);
			//if(a >= 144)
			if(a > 144 || a < 0)
			{
				possible = FALSE;
				break;
			}
			a = (a + y);
			//if(a > 144 || (a % x) != 0)
			if(a > 144 || a < 0 || (a % x) != 0)
			{
				possible = FALSE;
				break;
			}
			a = (a / x);
			//if(a >= 144)
			if(a > 144 || a < 0)
			{
				possible = FALSE;
				break;
			}
			a = (a - o);
			//if(a >= 144)
			if(a > 144 || a < 0)
			{
				possible = FALSE;
				break;
			}
			answer = a;
		}
		
		if(possible)
		{
			printf("The number left over is %d.\n", answer);
		}
		else
		{
			printf("The trick is too difficult to do.\n");
		}
	}
	return 0;
}

File: magic.java
import java.io.*;
import java.util.*;

public class magic 
{
	public static void main (String [] args) throws IOException
	{
		Scanner scan = new Scanner(System.in);
		
		int mult = scan.nextInt();
		int add = scan.nextInt();
		
		
		int z = 1;
		while (mult != 0 || add != 0)
		{
			boolean possible = true;
			for (int o = 1; o <= 10; o++)
			{
				if (mult == 0)
				{
					possible = false;
					break;
				}
					
				if (add % mult != 0) 
					possible = false;
					
				int cur = mult * o;
				
				if (cur > 144 || cur < 0)
					possible = false;
				
				cur += add;
				
				if (cur > 144 || cur < 0)
					possible = false;
					
				cur /= mult;
				
				if (cur > 144 || cur < 0)
					possible = false;
					
				cur -= o;
				
				if (cur > 144 || cur < 0)
					possible = false;
				
				
			}
			if (possible)
				System.out.printf("The number left over is %d.\n", add / mult);
			else
				System.out.println("The trick is too difficult to do.");
			
			mult = scan.nextInt();
			add = scan.nextInt();
		}
	}
}

